engines/wintermute/base/gfx/opengl/base_render_opengl3d.cpp

```diff
@@ -582,4 +525,159 @@ bool BaseRenderOpenGL3D::drawSpriteEx(BaseSurfaceOpenGL3D &tex, const Wintermute
 	return true;
 }
 
+void BaseRenderOpenGL3D::renderSceneGeometry(BaseArray<AdWalkplane *> &planes, BaseArray<AdBlock *> &blocks, BaseArray<AdGeneric *> &generics, Camera3D *camera) {
+	_gameRef->_renderer3D->resetModelViewTransform();
+	_gameRef->_renderer3D->setup3D(camera, true);
+
+	glDisable(GL_LIGHTING);
+	glDisable(GL_DEPTH_TEST);
+	glFrontFace(GL_CCW);
+	glPolygonMode(GL_FRONT_AND_BACK, GL_LINE);
+
+	glBindTexture(GL_TEXTURE_2D, 0);
+
+	// render walk planes
+	for (uint i = 0; i < planes.size(); i++) {
+		if (!planes[i]->_active) {
+			continue;
+		}
+
+		planes[i]->_mesh->render();
+
+		//m_Renderer->m_NumPolygons += _planes[i]->m_Mesh->m_NumFaces;
+	}
+
+	// render blocks
+	for (uint i = 0; i < blocks.size(); i++) {
+		if (!blocks[i]->_active) {
+			continue;
+		}
+
+		blocks[i]->_mesh->render();
+
+		//		m_Renderer->m_NumPolygons += _blocks[i]->m_Mesh->m_NumFaces;
+	}
+
+	// render generic objects
+	for (uint i = 0; i < generics.size(); i++) {
+		if (!generics[i]->_active) {
+			continue;
+		}
+
+		generics[i]->_mesh->render();
+
+		//		m_Renderer->m_NumPolygons += _generics[i]->m_Mesh->m_NumFaces;
+	}
+
+	_gameRef->_renderer3D->resetModelViewTransform();
+
+//	for (i = 0; i < _lights.size(); ++i) {
```

Comment:
debug code put into #if 0 or runtime condition

Reply:
Actually, the rendering of lights might be added back as well, only one parameter has to be added. Let me change this accordingly. The code below for waypoints could be implemented as well later.

Reply:
Light directions are rendered again now, the waypoints rendering code is removed, if necessary, it can be looked up from the original sources.